modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/metadata/MetadataExtensionFunctionalTestCase.java

```diff
@@ -168,6 +184,9 @@ ComponentMetadataDescriptor getSuccessComponentDynamicMetadata() {
     MetadataResult<ComponentMetadataDescriptor<T>> componentMetadata = getComponentDynamicMetadata(key);
     String msg = componentMetadata.getFailures().stream().map(f -> "Failure: " + f.getMessage()).collect(joining(", "));
     assertThat(msg, componentMetadata.isSuccess(), is(true));
+    if (!key.getId().equals(NULL_METADATA_KEY.getId())) {
```

Comment:
don't use the NULL_KEY to handle the assertion, if we have cases where the key has to be validated then handle it in each test, or create a `withKey` helper method that always asserts this